mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,170 +1,110 @@
+where
+    Foot: DataSet,
+    Land: DataSet,
+    Urban: DataSet,
+{
+    pub async fn is_ready(&self) -> bool {
+        self.urbanization.lock().await.is_ready()
+            && self.footfall.lock().await.is_ready()
+            && self.landtype.lock().await.is_ready()
     }
 }
 
-impl From<Landtype> for Assignment {
-    fn from(value: Landtype) -> Self {
-        match value {
-            Landtype::Built => Assignment::A,
-            //
-            Landtype::Tree => Assignment::B,
-            Landtype::Shrub => Assignment::B,
-            Landtype::Grass => Assignment::B,
-            //
-            Landtype::Bare => Assignment::C,
-            Landtype::Crop => Assignment::C,
-            Landtype::Frozen => Assignment::C,
-            Landtype::Water => Assignment::C,
-            Landtype::Wet => Assignment::C,
-            Landtype::Mangrove => Assignment::C,
-            Landtype::Moss => Assignment::C,
-        }
+impl<Foot, Land, Urban> HexBoostData<Foot, Land, Urban>
+where
+    Foot: HexAssignment,
+    Land: HexAssignment,
+    Urban: HexAssignment,
+{
+    pub async fn assignments(&self, cell: hextree::Cell) -> anyhow::Result<HexAssignments> {
+        let footfall = self.footfall.lock().await;
+        let landtype = self.landtype.lock().await;
+        let urbanization = self.urbanization.lock().await;
+        HexAssignments::from_data_sets(cell, &*footfall, &*landtype, &*urbanization)
     }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum Landtype {
-    Tree = 10,
-    Shrub = 20,
-    Grass = 30,
-    Crop = 40,
-    Built = 50,
-    Bare = 60,
-    Frozen = 70,
-    Water = 80,
-    Wet = 90,
-    Mangrove = 95,
-    Moss = 100,
+trait DiskTreeLike: Send + Sync {
+    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>>;
 }
 
-impl std::fmt::Display for Landtype {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str(self.to_str())
+impl DiskTreeLike for DiskTreeMap {
+    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>> {
+        self.get(cell)
     }
 }
 
-impl Landtype {
-    pub(crate) fn to_str(self) -> &'static str {
-        match self {
-            Landtype::Tree => "TreeCover",
-            Landtype::Shrub => "Shrubland",
-            Landtype::Grass => "Grassland",
-            Landtype::Crop => "Cropland",
-            Landtype::Built => "BuiltUp",
-            Landtype::Bare => "BareOrSparseVeg",
-            Landtype::Frozen => "SnowAndIce",
-            Landtype::Water => "Water",
-            Landtype::Wet => "HerbaceousWetland",
-            Landtype::Mangrove => "Mangroves",
-            Landtype::Moss => "MossAndLichen",
-        }
+impl DiskTreeLike for std::collections::HashSet<hextree::Cell> {
+    fn get(&self, cell: hextree::Cell) -> hextree::Result<Option<(hextree::Cell, &[u8])>> {
+        Ok(self.contains(&cell).then_some((cell, &[])))
     }
 }
```

Comment:
This trait appears redundant.
DiskTreeMap already has this interface. No other types are being used in place of a DiskTreeMap.

Reply:
I only wanted to grab the addition side of the diff, sorry.